### PR TITLE
EPMRPP-37807 || Seaweedfs plugin scale api

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/integration/plugin/impl/PluginLoaderImpl.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	}
+
+	@Override
+	public void deleteTempPlugin(String pluginFileDirectory, String pluginFileName) {
+		try {
+
+			Files.deleteIfExists(Paths.get(pluginFileDirectory, pluginFileName));
+
+		} catch (IOException e) {
+			//error during temp plugin is not crucial, temp files cleaning will be delegated to //TODO impl Quartz job to clean temp files
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/integration/plugin/impl/PluginLoaderImpl.java]: clean

File: src/main/java/com/epam/ta/reportportal/core/integration/plugin/impl/CreatePluginHandlerImpl.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+		this.pluginsRootPath = pluginsRootPath;
+		this.pluginBox = pluginBox;
+		this.pluginLoader = pluginLoader;
+		this.integrationTypeRepository = integrationTypeRepository;
+		this.dataStore = dataStore;
+		this.pluginUploadingCache = pluginUploadingCache;
+	}
+
+	@Override
+	public EntryCreatedRS uploadPlugin(MultipartFile pluginFile) {
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/integration/plugin/impl/CreatePluginHandlerImpl.java]: probably we need to create more private methods to clean the method up

File: src/main/java/com/epam/ta/reportportal/job/PluginLoaderService.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.List;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+public interface PluginLoaderService {
+
+	List<PluginInfo> getAllPluginsInfo(List<IntegrationType> integrationTypes);
+
+	List<PluginInfo> getNotLoadedPluginsInfo(List<IntegrationType> integrationTypes);
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/job/PluginLoaderService.java]: what does it mean - "notLoaded"?
Can we comment out all interfaces/their methods?

File: src/main/java/com/epam/ta/reportportal/core/integration/plugin/impl/PluginLoaderImpl.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/integration/plugin/impl/PluginLoaderImpl.java]: done

File: src/main/java/com/epam/ta/reportportal/core/integration/plugin/impl/CreatePluginHandlerImpl.java (hunk 2 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/integration/plugin/impl/CreatePluginHandlerImpl.java]: done

File: src/main/java/com/epam/ta/reportportal/core/integration/plugin/PluginLoader.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+	PluginInfo extractPluginInfo(Path pluginPath);
+
+	PluginState reloadPlugin(PluginWrapper plugin);
+
+	boolean validatePluginExtensionClasses(String pluginId);
+
+	Optional<PluginWrapper> retrieveOldPlugin(String newPluginId, String newPluginFileName);
+
+	void deleteOldPlugin(PluginWrapper oldPluginWrapper, String newPluginFileName);
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/integration/plugin/PluginLoader.java]: renamed, java docs were added

File: src/main/java/com/epam/ta/reportportal/core/integration/plugin/PluginLoader.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+	void createTempPluginsFolderIfNotExists(String path);
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/integration/plugin/PluginLoader.java]: moved to the private method

File: src/main/java/com/epam/ta/reportportal/job/PluginLoaderService.java (hunk 3 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/job/PluginLoaderService.java]: renamed. Java docs were provided.

File: src/main/java/com/epam/ta/reportportal/core/integration/plugin/PluginUploadingCache.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.stereotype.Component;
+
+import java.nio.file.Path;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+@Component
+public class PluginUploadingCache {
```
[the remark this answers is not in the document]
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/integration/plugin/PluginUploadingCache.java]: Component have been replaced by Cache in the Plugin manager

**Pull request merged**